openmsupply/open-msupply
# 2143 save as visited 2

Fixes #2143 

The following approach has been chosen: The UI stays mostly as before, only if the encounter is in the past and the status Pending a change has been introduced.
In this case the save button shows a confirmation menu if the encounter shouldn't be saved as "Visited":

![image](https://github.com/openmsupply/open-msupply/assets/88299195/bcc3edd0-b6f2-4595-a0e7-cf7e57ab9248)

Happy to change the Modal design...

# Testing

- Create an encounter in the future: save button should never trigger the new dialog
- Create an encounter in the past with status Pending: dialog should show when saving
- Create and encounter in the past with status != Pending: dialog shouldn't show
